extensions/liquibase-mongodb/deployment/src/main/java/io/quarkus/liquibase/mongodb/deployment/LiquibaseMongodbProcessor.java

```diff
@@ -289,25 +261,20 @@ private List<String> getChangeLogs(LiquibaseMongodbBuildTimeConfig liquibaseBuil
 
         ChangeLogParserFactory changeLogParserFactory = ChangeLogParserFactory.getInstance();
 
-        Set<String> resources = new LinkedHashSet<>();
+        try (var classLoaderResourceAccessor = new ClassLoaderResourceAccessor(
+                Thread.currentThread().getContextClassLoader())) {
 
-        ClassLoaderResourceAccessor classLoaderResourceAccessor = new ClassLoaderResourceAccessor(
-                Thread.currentThread().getContextClassLoader());
-
-        try {
-            resources.addAll(findAllChangeLogFiles(liquibaseBuildConfig.changeLog, changeLogParserFactory,
-                    classLoaderResourceAccessor, changeLogParameters));
+            Set<String> resources = new LinkedHashSet<>(
+                    findAllChangeLogFiles(liquibaseBuildConfig.changeLog, changeLogParserFactory,
+                            classLoaderResourceAccessor, changeLogParameters));
 
             LOGGER.debugf("Liquibase changeLogs: %s", resources);
 
             return new ArrayList<>(resources);
 
-        } finally {
-            try {
-                classLoaderResourceAccessor.close();
-            } catch (Exception ignored) {
-                // close() really shouldn't declare that exception, see also https://github.com/liquibase/liquibase/pull/2576
-            }
+        } catch (Exception ex) {
+            // close() really shouldn't declare that exception, see also https://github.com/liquibase/liquibase/pull/2576
+            throw new AssertionError(ex);
```

Comment:
Sorry I missed this but I think we need something better here.

Because the exception is caught globally not only for the close() now and it could happen in the code that is in the try. So I would rather have an IllegalStateException with a proper message and the cause.

Reply:
I changed the exception accordingly